Source/control.cpp

```diff
@@ -1663,40 +1673,37 @@ int CapStatPointsToAdd(int remainingStatPoints, const PlayerStruct &player, attr
 
 void ReleaseChrBtns(bool addAllStatPoints)
 {
-	int i;
-
 	chrbtnactive = false;
-	for (i = 0; i < 4; ++i) {
-		if (chrbtn[i]) {
-			chrbtn[i] = false;
-			if (MouseX >= ChrBtnsRect[i].x
-			    && MouseX <= ChrBtnsRect[i].x + ChrBtnsRect[i].w
-			    && MouseY >= ChrBtnsRect[i].y
-			    && MouseY <= ChrBtnsRect[i].y + ChrBtnsRect[i].h) {
-				PlayerStruct &player = plr[myplr];
-				int statPointsToAdd = addAllStatPoints ? player._pStatPts : 1;
-				switch (i) {
-				case 0:
-					statPointsToAdd = CapStatPointsToAdd(statPointsToAdd, player, attribute_id::ATTRIB_STR);
-					NetSendCmdParam1(true, CMD_ADDSTR, statPointsToAdd);
-					player._pStatPts -= statPointsToAdd;
-					break;
-				case 1:
-					statPointsToAdd = CapStatPointsToAdd(statPointsToAdd, player, attribute_id::ATTRIB_MAG);
-					NetSendCmdParam1(true, CMD_ADDMAG, statPointsToAdd);
-					player._pStatPts -= statPointsToAdd;
-					break;
-				case 2:
-					statPointsToAdd = CapStatPointsToAdd(statPointsToAdd, player, attribute_id::ATTRIB_DEX);
-					NetSendCmdParam1(true, CMD_ADDDEX, statPointsToAdd);
-					player._pStatPts -= statPointsToAdd;
-					break;
-				case 3:
-					statPointsToAdd = CapStatPointsToAdd(statPointsToAdd, player, attribute_id::ATTRIB_VIT);
-					NetSendCmdParam1(true, CMD_ADDVIT, statPointsToAdd);
-					player._pStatPts -= statPointsToAdd;
-					break;
-				}
+	for (int i = ATTRIB_STR; i <= ATTRIB_VIT; ++i) {
+		if (!chrbtn[i])
+			continue;
+
+		chrbtn[i] = false;
+		if (MouseX >= ChrBtnsRect[i].x
+			&& MouseX <= ChrBtnsRect[i].x + ChrBtnsRect[i].w
+			&& MouseY >= ChrBtnsRect[i].y
+			&& MouseY <= ChrBtnsRect[i].y + ChrBtnsRect[i].h) {
+			PlayerStruct &player = plr[myplr];
+			int statPointsToAdd = 1;
+			if (addAllStatPoints)
+				statPointsToAdd = CapStatPointsToAdd(player._pStatPts, player, (attribute_id)i);

```

Comment:
You really aren't a fan of the ternary assignment operator heh? The version without it look much less readable and more error prone to me.

Reply:
I actually love it. But i hate multiple statements in a single line. I fine that to be much more error prone.

I find it sad that C++ doesn't have the elvis variant (GCC does, but not other compilers)